onie-mk-demo.sh

```diff
@@ -89,6 +89,10 @@ output_raw_image=$(cat onie-image.conf | grep OUTPUT_RAW_IMAGE | cut -f2 -d"=")
 output_raw_image=$(eval echo $output_raw_image)
 
 # Tailor the demo installer for OS mode or DIAG mode
+if [ $sonic_kdump_enable = "y" ];
+then
+    sed -i "s/[^M] quiet/ crashkernel=256M quiet/" $tmp_installdir/install.sh
+fi
```

Comment:
You can use the variable GRUB_CMDLINE_LINUX instead of directly modifying script

Reply:
could you please elaborate on this suggestion?

Reply:
@qiluo-msft 
could you please elaborate on this suggestion?

Reply:
Sorry I made a wrong suggestion. You may consider the macro EXTRA_CMDLINE_LINUX, but install.sh need add it at the same time. `sed` a bash script is not recommended.

---
In reply to: [331246069](https://github.com/Azure/sonic-buildimage/pull/3276#discussion_r331246069) [](ancestors = 331246069)

Reply:
@qiluo-msft 

there are big chunk of code sedding the bash script next to my change, I am doing in very similar way

sed -i -e "s/%%ENABLE_KDUMP%%/$sonic_kdump_enable/g" \
       -e "s/%%DEMO_TYPE%%/$demo_type/g" \
       -e "s/%%IMAGE_VERSION%%/$image_version/g" \
       -e "s/%%ONIE_IMAGE_PART_SIZE%%/$onie_image_part_size/" \
       -e "s/%%EXTRA_CMDLINE_LINUX%%/$EXTRA_CMDLINE_LINUX/" \
    $tmp_installdir/install.sh || clean_up 1

Reply:
Please explorer this suggestion. The `sed` you mentioned is much conservative and safe.
> consider the macro EXTRA_CMDLINE_LINUX, but install.sh need add it at the same time

Reply:
@qiluo-msft 

please check my latest commit carefully, I am doing the same way as other sed but it still sed a script. EXTRA_CMDLINE_LINUX has been used already, I could not reuse it. if necessary , could we arrange a phone call to sort this out?

Reply:
@qiluo-msft 

thanks a lot for the review and approval:)